components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,958 @@
+
+                String logoutRedirectURI = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CALLBACK_URL);
+                paramMap.put(UAEPassAuthenticatorConstants.UAE.REDIRECT_URI, logoutRedirectURI);
+                String sessionID = context.getContextIdentifier() + "," + UAEPassAuthenticatorConstants.UAE.LOGIN_TYPE;
+                paramMap.put(UAEPassAuthenticatorConstants.UAE.OAUTH2_PARAM_STATE, sessionID);
+
+                String envUAEPass = getUAEPassEnvironment(context);
+                String logoutUrl = getLogoutUrl(envUAEPass);
+                logoutUrl = FrameworkUtils.buildURLWithQueryParams(logoutUrl, paramMap);
+                logoutUrl = modifyLogoutUrl(logoutUrl);
+                response.sendRedirect(logoutUrl);
+
+            } catch (IllegalArgumentException | IOException e) {
+                LOG.error("Error in initiate logout URI build.");
+                String idpName = context.getExternalIdP().getName();
+                String tenantDomain = context.getTenantDomain();
+                throw new LogoutFailedException("Error occurred while initiating the logout request to IdP: " + idpName
+                        + " of tenantDomain: " + tenantDomain, e);
+            }
+        } else {
+            super.initiateLogoutRequest(request, response, context);
+        }
+    }
+
+    /**
+     * After successful logout, WSO2 IS returns this response.
+     * Contains the details about the SP.
+     *
+     * @param request   The request that is received by the authenticator.
+     * @param response  The response that is received to the authenticator.
+     * @param context   The Authentication context received by authenticator.
+     */
+    @Override
+    protected void processLogoutResponse(HttpServletRequest request, HttpServletResponse response,
+                                         AuthenticationContext context) {
+
+        if (LOG.isDebugEnabled()) {
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                LOG.debug("Handled logout response from service provider " + request.getParameter("sp") +
+                        " in tenant domain " + IdentityTenantUtil.getTenantDomainFromContext());
+            } else {
+                LOG.debug("Handled logout response from service provider " + request.getParameter("sp") +
+                        " in tenant domain " + request.getParameter("tenantDomain"));
+            }
+        }
+    }
+
+    /**
+     * This method is used add additional parameters along with the authorize request.
+     *
+     * @param authenticatorProperties  The user input fields of the authenticator.
+     * @param loginPage                Current authorize URL.
+     * @return authzUrl                Returns the modified authorized URL appending the additional query params.
+     */
+    public String processAdditionalQueryParamSeperation(Map<String, String> authenticatorProperties, String loginPage)
```

Comment:
Do we need to keep this method as public

Reply:
Fixed with : [e41125306](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/e41125306054f4b7345ef374ad89a9ab2fdfea52)

Reply:
Fixed with : [c9f22571bc5](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/c9f22571bc5ffd21428c8b3a022c61e2045de40f)